benchmark/operations/read_request.py

```diff
@@ -2,21 +2,49 @@
 
 from twisted.internet.defer import succeed
 
+from flocker.apiclient import IFlockerAPIV1Client
+
 from .._interfaces import IProbe, IOperation
 
 
+class InvalidMethod(Exception):
+    """
+    Method not suitable for use with ReadRequest operation.
+
+    The method must be provided by the IFlockerAPIV1Client interface,
+    and require no parameters.
+    """
+
+
+def validate_method_name(interface, method_name):
+    """
+    Check that method name exists in interface and requires no parameters.
+
+    :param zope.interface.Interface interface: Interface to validate against.
+    :param str method_name: Method name to validate.
+    :raise InvalidMethod: if name is not valid or requires parameters.
+    """
+    for name, method in interface.namesAndDescriptions():
+        if name == method_name:
+            if len(method.getSignatureInfo()['required']) > 0:
+                raise InvalidMethod('Require no-arg method')
+            return
+    raise InvalidMethod(
+        'Method not found in interface {}'.format(interface.__name__)
+    )
```

Comment:
Please include `method_name` in the exception message.

Optionally, change `InvalidMethod` to take the interface & method as parameters and create the message there.